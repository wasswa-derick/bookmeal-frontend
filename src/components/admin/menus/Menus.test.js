import React from "react";
import { shallow } from "enzyme";
import { MenusPage } from "./Menus";

const err = {
  response: {
    status: 400
  }
};
const fn = () => Promise.reject(err);

describe("MenusPage", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MenusPage menus={[]} getMenus={fn} />);
  });

  it("should contain a link to set new menu page", () => {
    const linkProps = wrapper.find("Link").props();
    expect(linkProps.to).toBe("/admin/menus/new");
  });

  it("should render table", () => {
    expect(wrapper.find("table").length).toEqual(1);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
