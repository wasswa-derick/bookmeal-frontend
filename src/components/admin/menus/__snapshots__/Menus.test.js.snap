// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenusPage should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="stat"
      >
        <h2>
          0
           
          <i
            className="fa fa-bars"
          />
        </h2>
        <label
          htmlFor="menus"
        >
          Menus
        </label>
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="stat"
      >
        <Link
          className="btn btn-primary"
          href="/admin/menus/new"
          replace={false}
          to="/admin/menus/new"
        >
          Set New Menu
        </Link>
      </div>
    </div>
  </div>
  <hr />
  <section>
    <h4>
      Available Menus
    </h4>
    <hr />
    <Component
      component="div"
      loaded={false}
      loadedClassName="loadedContent"
      parentClassName="loader"
    >
      <table
        className="table table-striped"
        id="dtable"
      >
        <thead>
          <tr
            role="row"
          >
            <th>
              Menu Date
            </th>
            <th>
              Picture
            </th>
            <th>
              Meals
            </th>
            <th>
              Description
            </th>
            <th>
              Action
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </Component>
    <DeleteModal
      confirmDeletion={[Function]}
      title="menu"
    />
  </section>
</div>
`;
